src/backup.rs

```diff
@@ -26,22 +29,52 @@ struct Backup {
 
 
 impl BackupOptions {
-    pub fn backup(&self, archive_path: &Path, source: &Path, report: &Report) -> Result<()> {
+    pub fn backup(&self, archive_path: &Path, source: &Path, report: &Report, excludes: Option<Vec<&str>>) -> Result<()> {
```

Comment:
The `BackupOptions` is using the options builder pattern, which is a way to avoid having an over-long parameter set for `backup()`. Please set the excludes into the options into a field in `BackupOptions` from a separate method before running `backup`.